python-package/lightgbm/basic.py

```diff
@@ -408,7 +414,8 @@ def _choose_param_value(main_param_name: str, params: Dict[str, Any], default_va
     # avoid side effects on passed-in parameters
     params = deepcopy(params)
 
-    aliases = _ConfigAliases.get(main_param_name) - {main_param_name}
+    aliases = _ConfigAliases.get_sorted(main_param_name)
+    aliases = [a for a in aliases if a in params.keys() and a != main_param_name]
```

Comment:
Why this part `if a in params.keys()` is needed here?

Reply:
It was leftover from a previous implementation. I removed it in 4c4910f